common/defs.go

```diff
@@ -82,8 +108,261 @@ type PortUpdateMessage struct {
 	Port uint64 `json:"port"`
 }
 
+// Endpoint represents an endpoint - that is, something that
+// has an IP address and routes to/from. It can be a container,
+// a Kubernetes POD, a VM, etc.
+type Endpoint struct {
+	Any     string `json:"any,omitempty"`
+	Peer    string `json:"peer,omitempty"`
+	CidrStr string `json:"cidr,omitempty"`
+	// TODO this can be collapsed into Cidr but needs
+	// work on JSON marshaller/unmarshaller to do that.
+	Cidr              net.IPNet `json:"-"`
+	TenantID          uint64    `json:"tenant_id,omitempty"`
```

Comment:
TenantID and SegmentID are the database IDs we are using internally?

One day we will make those all strings, right?